lib/tds/types.ex

```diff
@@ -1708,8 +1713,11 @@ defmodule Tds.Types do
         <<fsec::little-unsigned-40>>
     end
   end
+
   def encode_time(time), do: encode_time(time, @max_time_scale)
+
   # DateTime2
+  @spec decode_datetime2(number, binary) :: NaiveDateTime.t()
```

Comment:
As mentioned in my  comment below (test case) decoder should return tuple so we can support 7 digit precision in case of custom ecto datetime type